api/audit/models.py

```diff
@@ -129,19 +122,4 @@ def update_environments_updated_at(self):
             updated_at=self.created_date
         )
 
-    def send_environments_to_dynamodb(self):
-        from environments.models import Environment
-
-        Environment.write_environments_to_dynamodb(
-            environment_id=self.environment_id, project_id=self.project_id
-        )
-
-    def send_environment_update_message(self):
-        if self.environment_id:
-            environment = self.environment
-            # Because we updated the environment `updated_at` in the previous hook in bulk
-            # update it manually here to save a `refresh_from_db` call
-            environment.updated_at = self.created_date
-            send_environment_update_message_for_environment(environment)
-        else:
-            send_environment_update_message_for_project(self.project)
+        process_environment_update.delay(args=(self.id,))
```

Comment:
Are we certain that `self.id` is populated at this point in the flow? I think it probably is but I'm not _certain_ that any of the tests fully exercise this?

Reply:
No, we have this: https://github.com/Flagsmith/flagsmith/pull/2807/files#diff-bf708a057d496be3362cd0687b441b41276bd6d62904e3346bbb1de29b09f497R175